Chat/Network.cpp

```diff
@@ -190,22 +187,20 @@ void Network::ProcessLogMessage(const LogMessage &msg, const std::string &ip)
             broadc_socket_.SendTo(answ_log_msg, msg_size, ip.c_str());
         }
         //do not put break here
-    case LOG_UPDATE:
-        user_ip_name_map_[ip] = msg.name_;
+    case LOG_UPDATE:;
+        ip_name_list_.Add(ip, msg.name_);
         break;
 
     case LOG_OFFLINE:
-        user_ip_name_map_.erase(ip);
+        ip_name_list_.Remove(ip);
         break;
     }
 }
 
 int Network::SendMsgTo(const std::string &user_name, const UserMsg &user_msg)
 {
-    std::map<std::string, std::string>::iterator it = std::find_if(user_ip_name_map_.begin(),
-                                                                   user_ip_name_map_.end(),
-                                                                   NameSearch(&user_name));
-    if(it == user_ip_name_map_.end())
+    std::string ip = ip_name_list_.GetIp(user_name);
+    if(!ip.size())
```

Comment:
it is better to use proper methods for such cases
http://www.cplusplus.com/reference/string/string/empty/